### ISSUE TITLE
Option to change background of lock-screen

### ISSUE THREAD

**Issue by cyb3rc0de** [mate-desktop/mate-screensaver#55]
It would be really, really nice to see option to change background of lock-screen that MATE has. This feature would fit perfectly inside "Desktop Settings" window, accessible from Control Center :)
##

**Comment by nachanon** [comment]
It seems that mate-screensaver's  screen lock always use system's default wallpaper image pointed by `/etc/alternatives/desktop-background`, and with _"Zoom"_ as wallpaper fitting policy.

So current workaround to set screen lock wallpaper is changing the symbolic link `/etc/alternatives/desktop-background` to point to the desired wallpaper file. Result looks like this:

![Screen lock with custom background wallpaper](https://cloud.githubusercontent.com/assets/5958794/6367977/e3710462-bd0a-11e4-86d5-504b5a27cee8.png)

Albeit this workaround works (with no need to logout or anything), it have a side effect of setting it as a default wallpaper for a new desktop environment, system-wide.

Note: `org.mate.background.picture-filename` dconf key @stefano-k mentioned does not seem to exist. And although another dconf key `org.mate.desktop.background.picture-filename` does exist, it affects (currently) only the desktop wallpaper, not the screen lock wallpaper.

MATE: MATE Screensaver 1.8.0, MATE 1.8.1 (from Debian Wheezy backports repository)
System: Debian GNU/Linux 7.0 Wheezy i386

**Comment by freed00m** [comment]
@nachanon I can confirm, the dconf values are the same for me in mint 17.1.

However I am not able to change it, because 
/usr/share/glib-2.0/schemas/000-system-background.gschema.override 
does not exist in mint 17.1
Any other ideas?

**Comment by stethewwolf** [comment]
hi everybody.
I use linuxmint 17.1 and i solve the problem changing the image linked at /usr/share/backgrounds/linuxmint/default_backgraund.jpg

**Comment by Lesik** [comment]
I can confirm that the dconf key `org.mate.background.picture-filename` does not exist.
Symlinking `/etc/alternatives/desktop-background` to the picture of your choice works.
Obviously those are hacky tricks and should not be the officially endorsed way of changing the background. A GUI for changing it should be developed.

Debian 8.0 Jessie, MATE 1.8.1

**Comment by dlussky** [comment]
I used symlinking, and it worked, but it's really mindblowing lack of basic option. It would be much more convenient if current desktop wallpaper was used. Or even better a dconf key for that. Sadly I'm completely unqualified to implement that, or I would definitely do it myself)

**Comment by nachanon** [comment]
Under mate-screensaver 1.8.0 function `gs_manager_init()` in `gs-manager.c` that runs at mate-screensaver daemon start, it seems that mate-screensaver tries to load system-default background using libmate-desktop's `mate_bg_load_from_system_preferences()`. Replacing that call with `mate_bg_load_from_preferences()` caused it to load user-configured background instead.

Also as the call was issued at daemon's startup, it would not see any background settings change until user logged-out and login again. So I moved the call from `gs_manager_init()` to `manager_show_window()` which runs at screen lock dialog display instead.

Here is a patch (for mate-screensaver 1.8.0):
[1004_use-user-background-on-screenlock-display.patch](https://gist.github.com/nachanon/834c705975124e38cb2a)

Following screenshot is captured after applied the patch:
![](https://cloud.githubusercontent.com/assets/5958794/8925856/56fca97a-3535-11e5-9ec3-6f16b30c398d.png)

You would see that screen lock background shown the same background as configured in `mate-appearance-properties`. Solid color, gradient, and image (with any fitting policy) works, although slideshow background didn't animate.

MATE: MATE Screen Saver 1.8.0, MATE 1.8.1 (patched against Debian Wheezy backports repository's source package)
System: Debian GNU/Linux 7.0 "Wheezy" i386

Note: Under current mate-screensaver git (1.10.2 revision 0ebac28), background loading seems to be done in a similar way. So, this is an **untested** patch for current mate-screensaver git:
[mate-screensaver_0ebac28_use-user-background-on-screenlock-display.patch](https://gist.github.com/nachanon/7f1aff255e24eb01b4fd)

**Comment by simonorono** [comment]
What's the status on this?

**Comment by eosrei** [comment]
@simonorono AFAIK this is it. You are looking at it. Potential implementations, but no pull request.

**Comment by cb474** [comment]
None of the workarounds here work for me. I used to just change the actual default image to whatever I want, but give it the name of the default image. That was /usr/share/backgrounds/mate/desktop/Stripes.png. But that seems to no longer be the default image. What and where is the default image now?

**Comment by irenge** [comment]
how to do it on Fedora 23 , i tried changing at /usr/share/desktop but did not change

**Comment by raveit65** [comment]
/usr/share/glib-2.0/schemas/mate-fedora.gschema.override
But a change will always overwritten by an update of mate-desktop package.

**Comment by kopias** [comment]
Could lockscren just use lightdm backgroun settings?

**Comment by TrendaFet** [comment]
Would love a GUI option for this as well, even if it just inherits the wallpaper would be something.

**Comment by JonasCz** [comment]
Would love to have this too - why not just use the desktop background as per @nachanon ?

**Comment by karlicoss** [comment]
The solution from https://github.com/mate-desktop/mate-screensaver/issues/55#issuecomment-156290334 almost worked for me, except for `/usr/share/glib-2.0/schemas/org.mate.background.gschema.xml` seems to be irrelevant to lock screen in Ubuntu Mate. The setting I had to change was `/usr/share/glib-2.0/schemas/ubuntu-mate.gschema.override`. Here's my simple script to set the lock screen:

```
#!/bin/bash
LOCK_SCREEN_BACKGROUND="/usr/share/backgrounds/ubuntu-mate-xenial/Blissful_Sky.jpg"
sudo crudini --set /usr/share/glib-2.0/schemas/ubuntu-mate.gschema.override org.mate.background picture-filename "'$LOCK_SCREEN_BACKGROUND'" # single qutes are important!
sudo glib-compile-schemas /usr/share/glib-2.0/schemas/
echo "Don't forget to restart X!"
```

[Crudini](https://github.com/pixelb/crudini) is not necessary (you can edit the `ubuntu-mate.gschema.override ` file yourself), but a really nice tool which handles editing ini files for you.

**Comment by ChrisCheney** [comment]
It appears for 18 it looks at /usr/share/backgrounds/linuxmint/default_background.jpg which is a symlink.

**Comment by rubencaro** [comment]
+1

**Comment by phillip-white-sociomantic** [comment]
Anybody find out how to resolve the zoom issue? Changing the wallpaper is fine, but it is zoomed in.

**Comment by fone** [comment]
is there a way to enable to default lock screen instead? the one you only see when you completely log out/login.

**Comment by charlesbos** [comment]
Overriding the system default for _org.mate.background picture-filename_ using a gschema override works but really mate-screensaver should respect the user's version of that setting just like gnome-screensaver and cinnamon-screensaver do for their equivalent keys. Please can this be implemented.

**Comment by nonsns** [comment]
+1

Symlink trick works system-wide, so I can live with it, although it does not play nice for multi-user installations (on the same pc).

Using the default bg just adds confusion for users with several mint boxes (they all look alike while they shouldn't)

Using the user-specified bg is really the way to go. Patch (which seems a clean solution) should be (tested more broadly and) upstreamed

**Comment by flexiondotorg** [comment]
Fixed via https://github.com/mate-desktop/mate-screensaver/commit/b2f48abf744a58f63cff97219e31532d361eb27f

**Comment by cb474** [comment]
I don't think that the patch b2f48ab actually fixed the feature request in this report. This should not be closed.

The request was for there to be a way for the user to set the background image of the lockscreen. The patch does not provide this, rather it forces the background of the lockscreen to match the user's desktop image. Hence the user still can not select the background image of the lockscreen specifically.

I don't want my lockscreen image to match my desktop. I want it to match the image for my login screen, which makes far more sense to me. As it is the patch breaks the workaround I had for this problem and makes it worse. It's very frustrating that after more than two years the solution is worse than the problem.

Please either get rid of the patch so that the old work around still works or make it so that you can actually specifically set the background of the lockscreen, independent of any other settings.

**Comment by raveit65** [comment]
> I don't think that the patch b2f48ab actually fixed the feature request in this report. This should not be closed.
> The request was for there to be a way for the user to set the background image of the lockscreen. The patch does not provide this, rather it forces the background of the lockscreen to match the user's desktop image. Hence the user still can not select the background image of the lockscreen specifically.

agree

> Please either get rid of the patch so that the old work around still works or make it so that you can actually specifically set the background of the lockscreen, independent of any other settings.

The commit is an improvement but a setting for changing the wallpaper is missing.

**Comment by cb474** [comment]
Thanks for reopening this feature request. Hopefully it will be less than two years before this gets addressed again. :)

**Comment by piramiday** [comment]
hi all, writing from the future -- March 2020. three years and still nothing?
any suggestions on how to deal with this and specify a custom lock screen background? :grin:

**Comment by shirishag75** [comment]
@raveit65 would this be packaged with mate 1.24.1 or the release after that, maybe a 1.24.2 ?

**Comment by raveit65** [comment]
Yes, it is a new feature and new features will be be packed with next major release.
But you can ask your distro maintainer to add it as patch to your current distro release.
Distro maintainer can do that easily :wink:

**Comment by evitalis** [comment]
> Yes, it is a new feature and new features will be be packed with next major release.
> But you can ask your distro maintainer to add it as patch to your current distro release.
> Distro maintainer can do that easily wink

So we are waiting on MATE 2.x for this, or did you mean minor version being 1.25 ?

I am not sure putting the onus on individual packagers is the right call. Its a lot of duplicated work for a patch that seems already merged here, in the central codebase.

I just upgraded to Fedora 32 and am still noticing the lock screen is different for some reason with no way to change it. This brought me back to this super old issue.

**Comment by raveit65** [comment]
Please check version of installed version of mate-screensaver. UI dialog of mate-screensaver-1.24.0-2.fc32 does have this feature since 3 month!!! https://src.fedoraproject.org/rpms/mate-screensaver/c/3dc0b7fe0febd353dbf3abf272dcd06d2b86fc78?branch=f32

**Comment by piramiday** [comment]
[Linux Mint 20 MATE](https://www.linuxmint.com/rel_ulyana_mate_whatsnew.php), based on Ubuntu 20.04 and released a couple of weeks ago with MATE 1.24, still doesn't include the fix:
```
user@mint20mate:~$ apt list mate-screensaver
Listing... Done
mate-screensaver/focal,now 1.24.0-1 amd64 [installed]
```

@raveit65 when do you think Ubuntu-based distros will be able to get it? thanks.

**Comment by raveit65** [comment]
I am fedora maintainer. No idea about debian/ubuntu/LinuxMint infrastructure and their plans. 
You need to file out a report at their bug reporting system and ask their maintainers.
